scripts/render-metrics-gif.py

```
@@ -34,163 +46,467 @@
+            "success": a["success"],
+            "all_found": has_det and len(info["missed"]) == 0,
+            "any_missed": has_det and len(info["missed"]) > 0,
+            "has_expectations": has_det,
+            "rules_found": sorted(info["found"]),
+            "rules_missed": sorted(info["missed"]),
+        })
+
+    return det_by_attack, phase_attacks
+
+
+def draw_frame(records, up_to, title, limits, width=1280, height=900):
```

Ruff (0.15.12)

[warning] Too many branches (30 > 12)

(PLR0912)